crates/typst-library/src/text/quotes.rs

```diff
@@ -180,7 +212,24 @@ impl<'s> Quotes<'s> {
             "no" | "nb" | "nn" if alternative => low_high,
             "ru" | "no" | "nb" | "nn" | "ua" => ("’", "’", "«", "»"),
             _ if lang.dir() == Dir::RTL => ("’", "‘", "”", "“"),
-            _ => return Self::default(),
+            _ => default,
+        };
+
+        let quotes = quotes.as_ref();
+        let [single_open, single_close] = match quotes
+            .map(|q| q.single.as_ref())
+            .flatten()
```

Comment:
`as_custom` takes ownership of `self` so we'd need an additional `clone` here. I know clones are cheap here, but we can completely skip the atomic increment on clone since we only borrow the quotes.

If we use `as_ref().as_custom()` then we have to do this for both levels:
```rust
let quotes = quotes.as_ref().as_custom();
let [single_open, single_close] = match quotes.and_then(|q| q.single.as_ref().as_custom()) {
    None => [single_open, single_close],
    Some(QuoteSet { open, close }) => [open, close].map(|s| s.as_str()),
};
```

At this point `and_then` on `Smart` is probably a good idea too, making it:
```rust
let quotes = quotes.as_ref();
let [single_open, single_close] = match quotes.and_then(|q| q.single.as_ref()) {
    Smart::Auto => [single_open, single_close],
    Smart::Custom(QuoteSet { open, close }) => [open, close].map(|s| s.as_str()),
};
```

Since `Smart` is essentially `Option` it probably doesn't hurt to mirror the API a bit more with things like `as_ref` and `and_then`.

Reply:
okay, it's fine to mirror the API a bit more then. I didn't see through the complications of my proposal. I just saw map + flatten, which clippy doesn't like because of and_then.